### PR TITLE
Service collaboration use cases

### PR DESCRIPTION
**Pull request by philecs**

### CODE REVIEW

File: docs/services/event-bus/CollaborationUseCases.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+Now let's discuss the individual services in details.
+
+#### Flow repository
+
+We need to extend the API by the following resources:
+
+- `POST /flows/{id}/start`: Used to start a flow
+- `POST /flows/{id}/stop`: Used to stop a flow
+
+Starting a flow means setting flow's `status` to `active` and raising `flow.starting` event. Stopping a flow means setting flow's `status` to `inactive` and raising `flow.stopping` event. The schema of the event payload is shown below.
```
Comment (pnedelko) [inline docs/services/event-bus/CollaborationUseCases.md]: Flow repo sets status `starting` and `stopping` accordingly upon receiving the HTTP calls. And upon receiving the `flow.started`/`flow.stopped` it changes the status to `active`/`inactive`.

File: docs/services/event-bus/CollaborationUseCases.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+Event:
+  type: object
+  required:
+    - headers
+    - payload
+  properties:
+    headers:
+      type: object
+       required:
+         - name
```
Comment (pnedelko) [inline docs/services/event-bus/CollaborationUseCases.md]: also `serviceName` field is there, which contains the name of the service, who sent the message.

File: docs/services/event-bus/CollaborationUseCases.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        createdAt:
+          type: string
+          format: date-time
+    payload:
+      type: object
+```
+
+The `payload` property is an arbitrary object to be sent with the event. Flow repository will send the entire flow as `payload`. 
+
+We also need to extend the `flow` model in the API by the following properties:
```
Comment (pnedelko) [inline docs/services/event-bus/CollaborationUseCases.md]: Seems like not relevant, cause already extended.

File: docs/services/event-bus/CollaborationUseCases.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+- `id`: required
+- `cron`: required for polling flows and defaults to `*/3 * * * *` if not set for a polling flow. Must not be set for a webhook flow.
+
+#### Webhooks
+
+Upon receiving `flow.starting` event the service checks if the `isWebhook` property is set. If so, the service persist a data record in his local DB but **doesn't start receiving HTTP requests** for the given flow yet. The following table demonstrates an example of such records.
```
Comment (pnedelko) [inline docs/services/event-bus/CollaborationUseCases.md]: if the `cron` property is **NOT** set

**Review by pnedelko — CHANGES_REQUESTED** [review]

File: docs/services/event-bus/CollaborationUseCases.md (hunk 2 above)
Comment (philecs) [reply docs/services/event-bus/CollaborationUseCases.md]: as required attribute?

Comment (pnedelko) [reply docs/services/event-bus/CollaborationUseCases.md]: @philecs yes

Comment (philecs) [reply docs/services/event-bus/CollaborationUseCases.md]: @pnedelko changed

**Comment by philecs** [comment]
Any other comments?

**Pull request merged**